pkg/actors/requests_responses.go

```diff
@@ -76,6 +78,24 @@ type ReminderResponse struct {
 	Period  string `json:"period"`
 }
 
+// MarshalJSON is a custom JSON marshaler that encodes the data as JSON.
+func (r *ReminderResponse) MarshalJSON() ([]byte, error) {
+	type responseAlias ReminderResponse
+	m := struct {
+		Data []byte `json:"data,omitempty"`
+		*responseAlias
```

Comment:
could you explain the need for this local type definition and the embedding here?

Reply:
This is how you implement a JSON marshaller that overrides a single field. We do this in other parts around Dapr and it's a common pattern in Go. Example: http://choly.ca/post/go-json-marshalling/